examples-util/util.rs

```diff
@@ -0,0 +1,28 @@
+#![allow(unused)]
+
+#[cfg(feature = "esp32")]
+pub use esp32_hal as hal;
+#[cfg(feature = "esp32c2")]
+pub use esp32c2_hal as hal;
+#[cfg(feature = "esp32c3")]
+pub use esp32c3_hal as hal;
+#[cfg(feature = "esp32c6")]
+pub use esp32c6_hal as hal;
+#[cfg(feature = "esp32s2")]
+pub use esp32s2_hal as hal;
+#[cfg(feature = "esp32s3")]
+pub use esp32s3_hal as hal;
```

Comment:
You can change the examples' Cargo.toml to include the hal as an alias. `hal = { package = ...`

Reply:
Does that work with multiple optional dependencies aliased to the same name? I think that's not valid toml

Reply:
ah - forget that - we have examples per chip 🤦‍♂️ yes ... good idea

Reply:
I tried this but workspace deps complicate things, it seems cargo doesn't understand this:

```
hal = { package = "esp32-hal", workspace = true }
```

and instead looks for the "hal" package `Caused by:
  dependency.hal was not found in `workspace.dependencies`.

Reply:
oh interesting ... cargo workspaces are always fun to use